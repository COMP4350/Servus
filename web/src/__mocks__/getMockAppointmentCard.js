function getMockAppointmentCard() {
    return {
        buyer: 'Risto',
        provider: 'Arveend',
        service_id: '1001',
        created_at: new Date(100),
        booked_time: new Date(500),
    }
}

export const mockAppointmentCard = getMockAppointmentCard();