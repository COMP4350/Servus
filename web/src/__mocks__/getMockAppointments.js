function getMockAppointments() {
    return [
        {
            buyer: 'Risto',
            provider: 'Arveend',
            service_id: '1001',
            created_at: new Date(100),
            booked_time: new Date(500),
        },
        {
            buyer: 'Arveend',
            provider: 'Risto',
            service_id: '1002',
            created_at: new Date(700),
            booked_time: new Date(1200),
        },
        {
            buyer: 'Caden',
            provider: 'Andy',
            service_id: '1003',
            created_at: new Date(1400),
            booked_time: new Date(1900),
        },
    ]
}

function getMockAppointmentsCookie() {
    return {
        username: 'testusername',
    }
}

export const mockAppointments = getMockAppointments();
export const mockAppointmentsCookie = getMockAppointmentsCookie();