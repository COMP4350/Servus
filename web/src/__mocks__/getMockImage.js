function getMockImage() {
    return {
        ownerUsername: 'riki',
        uploadDate: new Date(2012, 1, 1),
        imageName: 'example_image.jpeg',
        imageData: 'the/link/to/data',
        profilePicture: false,
    };
}

export const mockImage = getMockImage();
