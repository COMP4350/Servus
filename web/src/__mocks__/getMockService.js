function getMockService() {
    return {
        _id: '1001',
        name: "Risto's Painting",
        provider: 'Risto',
        description: 'Risto paints you',
        cost: 100,
        duration: '0100',
        availability: [{ weekday: 1, start_time: '0800', end_time: '1800' }],
        ratings: [{ username: 'arvind', rating: 5 }],
    };
}

export const mockService = getMockService();
