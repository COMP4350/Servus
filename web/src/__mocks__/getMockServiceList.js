function getMockServiceList() {
    return [
        {
            name: "Risto's Painting",
            provider: 'Risto',
            description: 'Risto paints you',
            cost: 100,
            duration: 60,
        },
        {
            name: "Reesto's Painting",
            provider: 'Reesto',
            description: 'Reesto paints yee',
            cost: 120,
            duration: 80,
        },
        {
            name: "Rasta's Painting",
            provider: 'Rasta',
            description: 'Rasta paints yaa',
            cost: 150,
            duration: 90,
        },
    ];
}

export const mockServiceList = getMockServiceList();
