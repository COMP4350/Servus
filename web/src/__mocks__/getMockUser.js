function getMockUser() {
    return {
        username: 'testuser',
        firstName: 'test',
        lastName: 'user',
        password: '123',
        bio: 'im a test user',
    };
}

export const mockUser = getMockUser();
