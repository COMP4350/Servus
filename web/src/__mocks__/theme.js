import { createMuiTheme } from '@material-ui/core/';

export default createMuiTheme({
    palette: {
        primary: {
            main: '#647AA3',
        },
        secondary: {
            main: '#EC5732',
        },
        default: {
            main: '#FFF2EB',
        },
    },
    background: {
        dark: '#151515',
        main: '#272727',
    },
    typographs: {
        fontFamily: ['Roboto'],
    },
});
