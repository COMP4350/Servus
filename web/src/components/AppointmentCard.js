import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { Card, CardContent, Typography, makeStyles } from '@material-ui/core';

const useStyles = makeStyles({
    date: {
        marginBottom: 12,
    },
    provider: {
        fontSize: 14,
    },
});

const AppointmentCard = props => {
    const classes = useStyles();
    const [service, setService] = useState();

    useEffect(() => {
        const fetchServiceInfo = async () => {
            const res = await axios.get(
                `${process.env.REACT_APP_API_HOST}/services/${props.service.id}`
            );
            setService(res.data.result);
        };
        fetchServiceInfo();
    }, [props]);

    return (
        <Card>
            <CardContent>
                <Typography color="textSecondary" className={classes.provider}>
                    {props.service.provider}
                </Typography>
                <Typography variant="h5">
                    {service ? service.name : null}
                </Typography>
                <Typography className={classes.date}>
                    {props.booked_time}
                </Typography>
                <Typography variant="body2" component="p">
                    {service ? service.description : null}
                </Typography>
            </CardContent>
        </Card>
    );
};

export default AppointmentCard;
