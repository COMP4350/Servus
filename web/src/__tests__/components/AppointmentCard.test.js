import React from 'react';
import { act, render } from '@testing-library/react';
import AppointmentCard from '../../components/AppointmentCard';
import { mockAppointmentCard } from '../../__mocks__/getMockAppointmentCard';
import mockAxios from 'axios';

describe('AppointmentCard', () => {
    it('renders correctly', async () => {
        const promise = Promise.resolve();
        let handleServiceUpdate = jest.fn(() => promise);
        let mockResult = mockAppointmentCard;

        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: { result: mockResult },
            })
        );

        const { asFragment } = render(
            <AppointmentCard 
                service={mockResult}
                setService={handleServiceUpdate}
            />
        )
        expect(asFragment()).toMatchSnapshot();
        expect(mockAxios.get).toHaveBeenCalledWith('/services/undefined');
        expect(mockAxios.get).toHaveBeenCalledTimes(1);

        await act(() => promise);
    });
});