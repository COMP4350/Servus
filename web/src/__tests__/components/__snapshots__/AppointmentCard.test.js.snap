// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root makeStyles-provider-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Arveend
      </p>
      <h5
        class="MuiTypography-root MuiTypography-h5"
      />
      <p
        class="MuiTypography-root makeStyles-date-1 MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body2"
      />
    </div>
  </div>
</DocumentFragment>
`;
